### PR TITLE
[OTA-3675] CTA Stepper

### PR DESCRIPTION
**Pull request by piotrsmol**
Signed-off-by: Piotr Smol <piotrsmol.pvt@gmail.com>

### CODE REVIEW

File: ota-plus-web/app/reactapp-new/src/components/dashboard/BuildTeam/styled.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import styled from 'styled-components';
+import { Icon, Button } from '../../common';
+
+const AddMembersButton = styled(Button)`
+  margin-bottom: 10px;
+  padding: 0;
+  font-weight: 500;
+  min-width: 110px;
+  span {
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/components/dashboard/BuildTeam/styled.ts]: maybe use '& > span' or we want to style all nested spans also?

File: ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+          } = generateStepData(index, primaryStatValue, secondaryStatValue);
+
+          return (
+            <DashboardStepperCard
+              status={
+                currentStep > index
+                  ? 'done'
+                  : currentStep === index
+                    ? 'active'
+                    : 'inactive'
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx]: maybe create a function like getStepStatus(currentStep, index)?

File: ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      </CardsWrapper>
+    </StepperWrapper>
+  );
+};
+
+const mapStateToProps = (state: AppState) => ({
+  campaigns: state.campaigns,
+  devices: state.devices,
+  software: state.software,
+  updates: state.updates
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx]: Let's create selectors maybe?

File: ota-plus-web/app/reactapp-new/src/store/campaigns/reducers.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+export const initialState: ICampaignsState = {
+  total: 0,
+  totalWithError: 0
+};
+
+export const campaignsReducer = (state = initialState, action: Actions): ICampaignsState => {
+  switch (action.type) {
+    case ActionType.SET_ALL_CAMPAIGNS_DONE: {
+      return {
+        ...state, ...action.payload
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/store/campaigns/reducers.ts]: saga returns { total, totalWithError } so we can just return { ...action.payload } if that's true for every case

File: ota-plus-web/app/reactapp-new/src/store/software/reducers.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+export const initialState: ISoftwareState = {
+  versionsTotal: 0
+};
+
+export const softwareReducer = (state = initialState, action: Actions): ISoftwareState => {
+  switch (action.type) {
+    case ActionType.SET_ALL_SOFTWARE_DONE: {
+      return {
+        ...state, ...action.payload
```
Comment (laniszew) [inline ota-plus-web/app/reactapp-new/src/store/software/reducers.ts]: I think we can just return action.payload here

File: ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx (hunk 2 above)
Comment (piotrsmol) [reply ota-plus-web/app/reactapp-new/src/components/dashboard/DashboardStepper/index.tsx]: Done

File: ota-plus-web/app/reactapp-new/src/store/campaigns/reducers.ts (hunk 4 above)
Comment (piotrsmol) [reply ota-plus-web/app/reactapp-new/src/store/campaigns/reducers.ts]: Done

File: ota-plus-web/app/reactapp-new/src/store/software/reducers.ts (hunk 5 above)
Comment (piotrsmol) [reply ota-plus-web/app/reactapp-new/src/store/software/reducers.ts]: Done

**Pull request merged**